pennylane/transforms/split_non_commuting.py

```diff
@@ -47,7 +100,10 @@ def shot_vector_postprocessing(results):
 
 @transform
 def split_non_commuting(
-    tape: QuantumScript, grouping_strategy: str | None = "default"
+    tape: QuantumScript,
+    grouping_strategy: str | None = "default",
+    shot_distribution: ShotDistributionFunction | str | None = None,
```

Comment:
We could also use `typing.Literal["uniform", "weighted", "weighted_random"]` here instead of just `str` to better communicate which strings are actually available.